packages/video_player/video_player/lib/video_player.dart

```diff
@@ -396,6 +434,33 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
     value = value.copyWith(volume: volume.clamp(0.0, 1.0));
     await _applyVolume();
   }
+
+  /// The closed caption based on the current [position] in the video.
+  ///
+  /// If there are no closed captions at the current [position], this will
+  /// return an empty [Caption].
+  ///
+  /// If no [closedCaptionFile] was specified, this will always return an empty
+  /// [Caption].
+  Caption _getCaptionAt(Duration position) {
+    if (_closedCaptionFile == null) {
+      return Caption();
+    }
+
+    // TODO: This would be more efficient as a binary search.
```

Comment:
Not a big deal either way, but for what it's worth I think you could trivially import this. https://api.flutter.dev/flutter/package-collection_collection/binarySearch.html

Reply:
Yes, it wouldn't be too hard, although we would have to either sort the captions at initialization, or make it clear that we assume a duration sorted list of captions (neither of which we currently do).

Reply:
Got it, I assumed these were already sorted. This is totally reasonable as-is.